tests/unit/type/zset.tcl

```diff
@@ -2480,4 +2480,29 @@ start_server {tags {"zset"}} {
         r zscore zz dblmax
     } {1.7976931348623157e+308}
 
+    test {zunionInterDiffGenericCommand acts on SET and ZSET} {
+        # Test one key is big and one key is small separately.
+        # The first one test zdiff, zdiffstore with listpack encoded set.
+        # The second one test zinter, zinterstore, zintercard, zdiff, zdiffstore with listpack encoded set.
+        foreach {big_key small_key zset_entries set_entries} {
+            zset{t} set{t} {1 a 2 b 3 c 4 d} {a b c}
+            set{t} zset{t} {1 a 2 b 3 c} {a b c d}
+        } {
```

Comment:
maybe add assert_encoding to make sure we hit what we aim for?
p.s. maybe i don't understand the big and small part, consider improving the comment.

Reply:
i will commit the test part first

Reply:
I updated the test as fully as possible according to my understanding, it should be able to cover many code paths.
although it looks a bit long and complicated... hope it can help